pkg/util/helper/taint_test.go

```diff
@@ -357,3 +359,283 @@ func TestGetNoExecuteTaints(t *testing.T) {
+		{
+			name: "the length of taints is 0",
+			args: args{
+				noExecuteTaints: nil,
+				usedTolerations: []corev1.Toleration{
+					{
+						Key:               "key",
+						Operator:          "Equal",
+						Value:             "value",
+						Effect:            "NoExecute",
+						TolerationSeconds: nil,
+					},
+				},
+			},
+			want: -1,
+		}, {
+			name: "the length of tolerations is 0",
+			args: args{
+				noExecuteTaints: []corev1.Taint{
+					{
+						Key:       "key",
+						Value:     "value",
+						Effect:    "NoExecute",
+						TimeAdded: nil,
+					},
+				},
+				usedTolerations: nil,
+			},
+			want: 0,
+		}, {
+			name: "test get minimal toleration time",
+			args: args{
+				noExecuteTaints: []corev1.Taint{
+					{
+						Key:       "key1",
+						Value:     "value1",
+						Effect:    "NoExecute",
+						TimeAdded: nil,
+					},
+					{
+						Key:       "key2",
+						Value:     "value2",
+						Effect:    "NoExecute",
+						TimeAdded: &timeAdded1,
+					},
+					{
+						Key:       "key3",
+						Value:     "value3",
+						Effect:    "NoExecute",
+						TimeAdded: &timeAdded2,
+					},
+				},
+				usedTolerations: []corev1.Toleration{
+					{
+						Key:               "key1",
+						Operator:          "Equal",
+						Value:             "value1",
+						Effect:            "NoExecute",
+						TolerationSeconds: nil,
+					},
+					{
+						Key:               "key2",
+						Operator:          "Equal",
+						Value:             "value2",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(10000),
+					},
+					{
+						Key:               "key3",
+						Operator:          "Equal",
+						Value:             "value3",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(20000),
+					},
+				},
+			},
+			want: 0,
+		}, {
+			name: "exist mismatch taint",
+			args: args{
+				noExecuteTaints: []corev1.Taint{
+					{
+						Key:       "key1",
+						Value:     "value1",
+						Effect:    "NoExecute",
+						TimeAdded: nil,
+					},
+					{
+						Key:       "key2",
+						Value:     "value2",
+						Effect:    "NoExecute",
+						TimeAdded: &timeAdded1,
+					},
+					{
+						Key:       "key3",
+						Value:     "value3",
+						Effect:    "NoExecute",
+						TimeAdded: &timeAdded2,
+					},
+				},
+				usedTolerations: []corev1.Toleration{
+					{
+						Key:               "key3",
+						Operator:          "Equal",
+						Value:             "value3",
+						Effect:            "NoExecute",
+						TolerationSeconds: nil,
+					},
+					{
+						Key:               "key4",
+						Operator:          "Equal",
+						Value:             "value4",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(10000),
+					},
+					{
+						Key:               "key5",
+						Operator:          "Equal",
+						Value:             "value5",
+						Effect:            "NoExecute",
+						TolerationSeconds: pointer.Int64(20000),
+					},
+				},
+			},
+			want: -1,
```

Comment:
This is not the expected value. The expected value should be 0. We need to modify the test method to meet this test case.